include/net/net_if.h

```diff
@@ -1362,6 +1362,17 @@ struct net_if_router *net_if_ipv4_router_add(struct net_if *iface,
 bool net_if_ipv4_addr_mask_cmp(struct net_if *iface,
```

Comment:
The commit message could better state that the interface selection happens via the assigned subnet as Zephyr does not support generic routes to be set. Then this addition is understandable that it looks at the netmask o fthe interface only.

Reply:
Yes, that makes sense.